### PR TITLE
Code quality

### PR DESCRIPTION
**Pull request by shagren**
I have added black, isort, mypy, flake8.
Now module developers can reformat and lint code before commit.
Also I add required typehinting and fix few problems with code

### CODE REVIEW

File: pyk4a/pyk4a.py [hunk 1]
```diff
@@ -82,150 +92,192 @@ def _stop_imu(self):
...
             :param timeout: Timeout in ms. Default is infinite.
 
         Returns:
-            :return CaptureResult containing requested images and infos if they are available in the current
-                capture. There are no guarantees that the returned object will contain all the requested images.
+            :return capture containing requested images and infos if they are available
+                in the current capture. There are no guarantees that the returned
+                object will contain all the requested images.
 
-        See default request in CaptureRequest
```
Comment (shagren) [inline pyk4a/pyk4a.py:97]: Like as this comment not relevant more. Removed.

File: pyk4a/pyk4a.py [hunk 2]
```diff
@@ -82,150 +92,192 @@ def _stop_imu(self):
...
+        res, capture_capsule = k4a_module.device_get_capture(
+            self._device_id, self.thread_safe, timeout
+        )
         self._verify_error(res)
 
         capture = PyK4ACapture(device=self, capture_capsule=capture_capsule)
         return capture
 
-    def get_imu_sample(self, timeout=TIMEOUT_WAIT_INFINITE):
-        res, imu_sample = k4a_module.device_get_imu_sample(self._device_id, self.thread_safe, PyK4A.TIMEOUT_WAIT_INFINITE)
```
Comment (shagren) [inline pyk4a/pyk4a.py:108]: There was missed `timeout` argument

File: pyk4a/pyk4a.py [hunk 3]
```diff
@@ -82,150 +92,192 @@ def _stop_imu(self):
...
+        res = k4a_module.device_set_color_control(
+            self._device_id, self.thread_safe, cmd, mode, value
+        )
         self._verify_error(res)
 
     @property
     def brightness(self) -> int:
         return self._get_color_control(ColorControlCommand.BRIGHTNESS)[0]
 
+    @brightness.setter
```
Comment (shagren) [inline pyk4a/pyk4a.py:168]: mypy requirement:   `setter` must follows `property`

File: pyk4a/pyk4a.py [hunk 4]
```diff
@@ -252,49 +306,65 @@ def _verify_error(res):
 
 
 class PyK4ACapture:
+    def __init__(self, device: PyK4A, capture_capsule: object):
```
Comment (shagren) [inline pyk4a/pyk4a.py:309]: moved `__init__` at start.

File: pyk4a/config.py [hunk 5]
```diff
@@ -85,13 +88,27 @@ def __init__(self,
...
-        return (self.color_format,
-                self.color_resolution,
-                self.depth_mode,
-                self.camera_fps,
-                self.synchronized_images_only,
-                self.depth_delay_off_color_usec,
-                self.wired_sync_mode,
-                self.subordinate_delay_off_master_usec,
-                self.disable_streaming_indicator)
+    def unpack(
```
Comment (lpasselin) [inline pyk4a/config.py:91]: Is there anyway to make this less ugly without cheating the formatting tools?

**Comment by lpasselin** [comment]
I think this is a great idea and if easy to implement, we should add a free CI service to automatically run the formatting scripts on pull requests

File: example/color_formats.py [hunk 6]
```diff
@@ -52,9 +52,9 @@ def convert_to_bgra_if_required(k4a, img_color):
     nbytes_NV12 = get_color_image_size(config_NV12, imshow=imshow)
     nbytes_YUY2 = get_color_image_size(config_YUY2, imshow=imshow)
 
-    print(f"{nbytes_BGRA32=} {nbytes_MJPG=}")
```
[the remark this answers is not in the document]
Comment (lpasselin) [reply example/color_formats.py:55]: This is valid. It prints the variable name followed by the value.
See https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

[the remark this answers is not in the document]
Comment (lpasselin) [reply example/color_formats.py:55]: I agree it is better to remove it for older version compatibility.

**Comment by lpasselin** [comment]
Also might not be worth it be is there an easy way to add formatting to the cpp file?

File: setup.cfg [hunk 7]
```diff
@@ -1,2 +1,26 @@
 [metadata]
 description-file = README.md
+
+
+[flake8]
+max-line-length = 88
```
Comment (lpasselin) [inline setup.cfg:6]: I know this is very personal but if we change this from the default, can we use 120? like [pytorch](https://github.com/pytorch/pytorch/blob/master/.flake8). I feel it is much easier to read, especially using github code review.

Comment (shagren) [reply setup.cfg]: Why not :)
Changed

File: pyk4a/config.py (hunk 5 above)
Comment (shagren) [reply pyk4a/config.py:91]: I made line-lenght 120 chars. Now it looks better

**Comment by shagren** [comment]
> I think this is a great idea and if easy to implement, we should add a free CI service to automatically run the formatting scripts on pull requests
> edit: maybe reuse the [kinect azure SDK pipeline file](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/blob/develop/azure-pipelines.yml)? Or reuse their artifacts? I would prefer working with docker though.

I will add  simple CI later

**Comment by shagren** [comment]
> Also might not be worth it be is there an easy way to add formatting to the cpp file?

Will check later. In any case it will be separate PR.

**Comment by lpasselin** [comment]
@shagren is this one ready to merge?

**Comment by shagren** [comment]
> @shagren is this one ready to merge?

@lpasselin, yes

**Pull request merged**